src/framework/application.js

```diff
@@ -2318,6 +2313,20 @@ var makeTick = function (_app) {
         application.fire("framerender");
 
         if (application.autoRender || application.renderNextFrame) {
+            // In AUTO mode the resolution is changed to match the canvas size
+            if (application._resolutionMode === RESOLUTION_AUTO) {
+                // Check if the canvas DOM has changed size
+                const canvas = application.graphicsDevice.canvas;
+                const pixelRatio = Math.min(application.graphicsDevice.maxPixelRatio, window.devicePixelRatio);
+
+                const clientWidth = canvas.clientWidth;
+                const clientHeight = canvas.clientHeight;
+
+                if (canvas.width !== Math.floor(clientWidth * pixelRatio) || canvas.height !== Math.floor(clientHeight * pixelRatio)) {
```

Comment:
shouldn't this test live in resizeCanvas instead of here? Calculating pixelRatio and doing all the floor stuff twice ain't great.

Reply:
I didn't want add this logic in resizeCanvas as it's existing API just in case some users were using the function directly.

I could add a new function that would be along the lines of 'checkAndResizeCanvas` or have a utility function in graphicDevice that has the logic for calculating the width/height?

Reply:
well actually the function already has the test? (see https://github.com/playcanvas/engine/pull/3124/files#diff-f7773f4bc0740b6db3da2ac63c30e595a6076b3f8d6699f5c0de9c8628ae7d57R3501). I don't think you need to do any calculation or testing here.

Reply:
So I'd suggest just making it something like:
```javascript
            // In AUTO mode the resolution is changed to match the canvas size
            if (application._resolutionMode === RESOLUTION_AUTO) {
                const canvas = application.graphicsDevice.canvas;
                application.graphicsDevice.resizeCanvas(canvas.clientWidth, canvas.clientHeight);
            }
```

Reply:
Completely makes sense now looking back on it 😅 

Made the changes.